### ISSUE TITLE
De-Fusion: WP REST API endpoints shared with wpcom

### ISSUE THREAD

**Issue by sdixon194** [Automattic/jetpack#25200]
### Impacted plugin

Jetpack

### What

WP REST API endpoints sharted with wpcom

### How

```
"$plugin_root/_inc/lib/core-api/wpcom-endpoints/" => "$wpcom_root/wp-content/rest-api-plugins/jetpack-endpoints/",
"$plugin_root/_inc/lib/core-api/wpcom-fields/"    => "$wpcom_root/wp-content/rest-api-plugins/jetpack-fields/",
```

**Comment by sdixon194** [comment]
List of all individual files:

```
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/business-hours.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-google-docs.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-mailchimp.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-podcast-player.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-publicize-share-post.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-resolve-redirect.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-search.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-transient.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-tweetstorm-gather.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-tweetstorm-parse.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/gutenberg-available-extensions.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/hello.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/memberships.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/publicize-connection-test-results.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/publicize-connections.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/publicize-services.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/service-api-keys.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/sites-posts-featured-media-url.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/subscribers.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php
```

Out of sync (by a potentially negligible amount):

```
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/service-api-keys.php
```

Doesn't exist on wpcom:

```
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-search.php
projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/subscribers.php
```

`projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/subscribers.php` seems to exist at `wp-content/rest-api-plugins/endpoints/subscribers.php` instead and is very out of sync.

**Comment by anomiex** [comment]
> projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php

D81254-code was never brought back to Jetpack for some reason.

> projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php

Looks like this one has been brought in sync?

> projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/service-api-keys.php

Negligible.

> `projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/subscribers.php` seems to exist at `wp-content/rest-api-plugins/endpoints/subscribers.php` instead and is very out of sync.

Indeed. I didn't dig too deeply, but it looks to me like the Jetpack version calls to wpcom while the wpcom version does things directly. Let's not worry about this one.

**Comment by sdixon194** [comment]
Defusion: D91618-code

I think D81254-code might need to be brought to Jetpack before we can merge this one though.

**Comment by anomiex** [comment]
> I think D81254-code might need to be brought to Jetpack before we can merge this one though.

Probably would be a good idea

**Comment by sdixon194** [comment]
Jetpack PR to bring in sync: https://github.com/Automattic/jetpack/pull/27258, once merged this will have to wait for 11.6-a.3 before merging.

**Comment by samiff** [comment]
D91618-code landed in r256068-wpcom
